src/base/utils/net.cpp

```diff
@@ -38,5 +40,42 @@ namespace Utils
         {
             return !QHostAddress(ip).isNull();
         }
+
+        bool isValidSubnet(const QPair<QHostAddress, int> &subnet)
```

Comment:
This function handles only one specific case of invalid subnet so it shouldn't be in Utils. BTW, it has small piece of trivial code so you can use it inplace.